File: Kotlin/XorEngine.kt
fun parity(x: Int): Int {
    var y = x xor (x shr 1)
    y = y xor (y shr 2)
    y = y xor (y shr 4)
    y = y xor (y shr 8)
    y = y xor (y shr 16)
    return y and 1
}

fun main(args: Array<String>) {
    for (t in 1..readLine()!!.toInt()) {
        var ctrE = 0
        var ctrO = 0
        val (N, Q) = readLine()!!.split(" ").map { it.toInt() }
        val A = readLine()!!.split(" ").map { it.toInt() }
        for(i in 1..Q){
            val P = readLine()!!.toInt()
            A.forEach {
                if(parity(it xor P) == 0)
                    ctrE++
                else
                    ctrO++
            }
            println("$ctrE $ctrO")
        }
    }
}

File: Python/XorEngine.py
from sys import stdin, stdout


def par_check(n):
    n = bin(int(n)).count('1')
    return n % 2


if __name__ == '__main__':
    for _ in range(int(input())):
        N, Q = map(int, input().split())
        A = list(map(par_check, stdin.readline().split()))
        ctrE = A.count(0)
        ctrO = A.count(1)
        for _ in range(Q):
            P = int(input())
            if par_check(P) == 0:
                stdout.write("%d %d\n" % (ctrE, ctrO))
            else:
                stdout.write("%d %d\n" % (ctrO, ctrE))
